### PR TITLE
feat: [CDS-48581]: Support for Tas App resize, Rollback, Swap Routes, Swap Rollback Step

### PR DESCRIPTION
**Pull request by sourabh-awashti**
## Describe your changes
Support for Tas App Resize, App Rollback, Swap Routes and Swap rollback steps manager side.
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/41635)

### CODE REVIEW

**Comment by yogesh-chauhan** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java` line 1053 at r8](https://reviewable.io/reviews/harness/harness-core/41635#-NJPDU5gA0Z0ZCA9WT-R:-NJPDU5gA0Z0ZCA9WT-S:b-q22rc8) ([raw file](https://github.com/harness/harness-core/blob/7f5f232eef9448d25dc0d8813b748e438d85e5c8/125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java#L1053)):*
> ```Java
>             .setName("App Rollback")
>             .setType(StepSpecTypeConstants.TAS_ROLLBACK)
>             .setStepMetaData(StepMetaData.newBuilder().addCategory("TAS").setFolderPath("TAS").build())
> ```

"TAS" is a string at some places and enum at others, is it expected ?

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java [hunk 1]
```diff
@@ -249,6 +257,8 @@ public class CDNGPlanCreatorProvider implements PipelineServiceInfoProvider {
   private static final String ELASTIGROUP = "Elastigroup";
   private static final String TAS = "TAS";
 
+  private static final String TAS = "TAS";
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java:260]: should be removed

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java [hunk 2]
```diff
@@ -1005,6 +1023,36 @@ public List<StepInfo> getStepInfo() {
...
+            .build();
+
+    StepInfo tasSwapRollback =
+        StepInfo.newBuilder()
+            .setName("Swap Rollback")
+            .setType(StepSpecTypeConstants.SWAP_ROLLBACK)
+            .setStepMetaData(StepMetaData.newBuilder().addCategory("TAS").setFolderPath("TAS").build())
+            .setFeatureFlag(FeatureName.CDS_TAS_NG.name())
+            .build();
+    StepInfo tasAppResizeRoutes =
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java:1041]: should be `tasAppResize`

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java [hunk 3]
```diff
@@ -1005,6 +1023,36 @@ public List<StepInfo> getStepInfo() {
...
+            .setStepMetaData(StepMetaData.newBuilder().addCategory(TAS).addFolderPaths(TAS).build())
+            .setFeatureFlag(FeatureName.CDS_TAS_NG.name())
+            .build();
+
+    StepInfo tasRollback =
+        StepInfo.newBuilder()
+            .setName("App Rollback")
+            .setType(StepSpecTypeConstants.TAS_ROLLBACK)
+            .setStepMetaData(StepMetaData.newBuilder().addCategory("TAS").setFolderPath("TAS").build())
+            .setFeatureFlag(FeatureName.CDS_TAS_NG.name())
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java:1054]: should be variable

File: 125-cd-nextgen/src/main/java/io/harness/cdng/pipeline/CDStepInfo.java [hunk 4]
```diff
@@ -94,6 +96,7 @@
         EcsRunTaskStepInfo.class, ElastigroupDeployStepInfo.class, ElastigroupRollbackStepInfo.class,
         ElastigroupSetupStepInfo.class, TerragruntPlanStepInfo.class, TerragruntApplyStepInfo.class,
         TerragruntDestroyStepInfo.class, TerragruntRollbackStepInfo.class, AsgCanaryDeployStepInfo.class,
+        AsgCanaryDeleteStepInfo.class, TasSwapRoutesStepInfo.class, TasSwapRollbackStepInfo.class,
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/pipeline/CDStepInfo.java:99]: should have all 4 steps

**Comment by sourabh-awashti** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java` line 1053 at r8](https://reviewable.io/reviews/harness/harness-core/41635#-NJPDU5gA0Z0ZCA9WT-R:-NJPhujO6vgVn5sGt2GB:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7f5f232eef9448d25dc0d8813b748e438d85e5c8/125-cd-nextgen/src/main/java/io/harness/cdng/creator/CDNGPlanCreatorProvider.java#L1053)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

"TAS" is a string at some places and enum at others, is it expected ?
</blockquote></details>

Done.

**Comment by abhayganvir** [comment]
trigger poccheckstyle

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 12 of 30 files at r3, 1 of 31 files at r4, 47 of 62 files at r5, 5 of 9 files at r6, 4 of 13 files at r7, 17 of 23 files at r9, 6 of 7 files at r10, 1 of 1 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41635)* status: 92 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained.

**Review by yogesh-chauhan — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41635)* status: 92 file(s) to review. 1 of 2 approvals obtained.

**Review by piyush-bhuwalka — DISMISSED** [review]
> Reviewed 12 of 30 files at r3, 1 of 31 files at r4, 12 of 62 files at r5, 5 of 9 files at r6, 3 of 13 files at r7, 17 of 23 files at r9, 5 of 7 files at r10, 1 of 1 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41635)* status: 36 file(s) to review.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 2 of 2 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41635)* status: 38 file(s) to review. 1 of 2 approvals obtained.

**Review by piyush-bhuwalka — APPROVED** [review]
> Reviewed 2 of 2 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41635)* status: 36 file(s) to review.

**Review by anilchowdhury-harness — APPROVED** [review]
:lgtm:

> Reviewed 12 of 30 files at r3, 1 of 31 files at r4, 47 of 62 files at r5, 5 of 9 files at r6, 13 of 23 files at r9, 5 of 7 files at r10, 1 of 1 files at r11, 1 of 2 files at r12, 17 of 17 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41635#-:-NJQojgv5FAuEVrdWAvC:bnfp4nl)* status: 17 file(s) to review. 1 of 2 approvals obtained.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 17 of 17 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41635)* status: :shipit: complete! ✓

**Comment by sourabh-awashti** [comment]
trigger ti1

**Comment by sourabh-awashti** [comment]
trigger ti1

**Comment by RishabhGupta34** [comment]
trigger tin1

**Pull request merged**